### PR TITLE
CLN: Assorted Cleanups

### PR DESCRIPTION
**Pull request by jbrockmendel**
- make PTA/DTA strftime() return ndarray instead of Index
- fix a xfailed Series[Period] test
- avoid imports of Index subclass in arrays.categorical and arrays.datetimelike
- some typing

### CODE REVIEW

File: pandas/tests/series/test_period.py [hunk 1]
```diff
@@ -71,10 +71,9 @@ def test_NaT_scalar(self):
         series[2] = val
         assert pd.isna(series[2])
 
-    @pytest.mark.xfail(reason="PeriodDtype Series not supported yet")
     def test_NaT_cast(self):
         result = Series([np.nan]).astype("period[D]")
-        expected = Series([pd.NaT])
+        expected = Series([pd.NaT], dtype="period[D]")
```
Comment (gfyoung) [inline pandas/tests/series/test_period.py]: This is changing a test, no?  Why are we dropping the `xfail` ?

**Review by jorisvandenbossche — COMMENTED** [review]
Nice clean-up, added few comments

File: pandas/core/arrays/categorical.py [hunk 2]
```diff
@@ -2725,17 +2724,15 @@ def _factorize_from_iterable(values):
...
     """
-    from pandas.core.indexes.category import CategoricalIndex
-
     if not is_list_like(values):
         raise TypeError("Input must be list-like")
 
-    if is_categorical(values):
-        values = CategoricalIndex(values)
-        # The CategoricalIndex level we want to build has the same categories
+    if is_categorical_dtype(values):
```
Comment (jreback) [inline pandas/core/arrays/categorical.py]: IIRC @jschendel didn't we just change this to make a perf fix? is this change compat with that

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/datetimelike.py [hunk 3]
```diff
@@ -1018,9 +1016,9 @@ def _add_delta_tdi(self, other):
 
         if isinstance(other, np.ndarray):
             # ndarray[timedelta64]; wrap in TimedeltaIndex for op
-            from pandas import TimedeltaIndex
+            from pandas.core.arrays import TimedeltaArray
 
-            other = TimedeltaIndex(other)
+            other = TimedeltaArray._from_sequence(other)
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py]: it is timedelta64, but not necessarily timedelta64[ns].

File: pandas/tests/series/test_period.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/series/test_period.py]: IIUC the xfail was from a time before `dtype="period[D]"` was recognized

File: pandas/core/arrays/datetimelike.py [hunk 4]
```diff
@@ -180,9 +180,7 @@ def strftime(self, date_format):
                'March 10, 2018, 09:00:02 AM'],
               dtype='object')
         """
-        from pandas import Index
-
-        return Index(self._format_native_types(date_format=date_format))
+        return self._format_native_types(date_format=date_format).astype(object)
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py]: otherwise it comes back with string-dtype

[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply pandas/core/arrays/datetimelike.py]: It already seems to give object dtype?

```
In [9]: pd.date_range("2012", periods=3).array._format_native_types()                                                                                         
Out[9]: array(['2012-01-01', '2012-01-02', '2012-01-03'], dtype=object)
```

[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py]: If you pass a format to strftime it'll come back with str-dtype

File: pandas/core/arrays/categorical.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py]: I put together this version by looking at the old version and taking out the non-executed paths.  So this shouldn't take a hti

Comment (jschendel) [reply pandas/core/arrays/categorical.py]: @jreback : I think you're referring to #27669, which should not be impacted by this.

Comment (jreback) [reply pandas/core/arrays/categorical.py]: OT: we *might* be able to remove is_categorical entirely?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

File: pandas/core/arrays/datetimelike.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply pandas/core/arrays/datetimelike.py]: @jbrockmendel I think we should avoid using `_from_sequence` for that in our code base. Maybe `array(.., dtype='timedelta64[ns])` instead ?

[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py]: why do you want to avoid _from_sequence?

File: pandas/core/arrays/categorical.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py]: Yah we should.  Will need to be deprecated since its in the API

**Pull request merged**